### DISCUSSION TITLE
Limit users on a self hosted instance

### DISCUSSION THREAD

**Discussion by rhyst** [anyproto/contributors#193] [Self-hosting]
In the self hosted docs it is stated:

```
any-sync networks are permissionless. This means that anyone who knows the IP and port of the coordinator node can connect to it, create new spaces, or download existing encrypted objects. Data inside spaces is always secure, as it requires space encryption keys to read it. Nodes do not store and do not have access to these encryption keys and cannot decrypt the data. Refer to the [Any-Sync Protocol overview](https://tech.anytype.io/any-sync/overview) for additional details.
```

Does this mean that if you run the sync components self hosted that there is no built in way to limit who can sign up to use the instance?

I suppose you can probably set the listen addresses such that the various components only listen on some VPN interface or something but that is a a reasonable amount extra work on top of a complicated set up already.

Thanks :)

**Comment by fuksman** [comment]
Hello @rhyst,

Technically, everyone who has a network configuration can create identities and spaces in this network. We have seen requests to implement restrictions for this and see how we can address this, but it is not our current priority.

In reality, it is unlikely that anyone has access to your network configuration unless you share it publicly.

**Reply by tribals** [reply]
> it is unlikely that anyone has access to your network configuration

Is is so due to the need to know "network public key" (or what is this actually, I don't know) in advance?

I mean, for the client app there is no way to connect to network knowing only it's address(es)/port and no way to retrieve "network public key" over network itself?
